### ISSUE TITLE
`mask` or `clip-path` clips filter effect outside object bounding box

### ISSUE THREAD

**Issue by stanio** [weisJ/jsvg#84]
<details>
<summary><code>filter-mask.svg</code></summary>

```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256">
  <defs>
    <filter id="drop-shadow" x="0" y="0">
      <?feDropShadow dx="5" dy="10" stdDeviation="2" flood-opacity="0.5" ?>
      <feGaussianBlur in="SourceAlpha" stdDeviation="2" />
      <feOffset dx="5" dy="10" result="offset-blur" />
      <feFlood flood-color="black" flood-opacity="0.5" />
      <feComposite in2="offset-blur" operator="in" />
      <!--feComposite in="SourceGraphic" operator="over" /-->
    </filter>
    <filter id="white-fill" x="0" y="0" width="1" height="1">
      <feFlood flood-color="white" />
      <feComposite in2="SourceAlpha" operator="in" />
    </filter>
    <mask id="graphic1-mask" x="0" y="0">
      <use href="#graphic1" filter="url(#white-fill)" />
    </mask>
    <clipPath id="graphic1-clip">
      <use href="#graphic1" />
    </clipPath>
  </defs>

  <rect id="graphic1" x="13" y="13" width="120" height="230" fill="green" />

  <!--use href="#graphic2" filter="url(#drop-shadow)" clip-path="url(#graphic1-clip)" /-->
  <use href="#graphic2" filter="url(#drop-shadow)" mask="url(#graphic1-mask)" />

  <circle id="graphic2" cx="120" cy="110" r="70" fill="orange" />
  <!--g id="graphic2">
    <rect x="50" y="40" width="160" height="160" opacity="0" />
    <circle cx="120" cy="110" r="70" fill="orange" />
  </g-->

</svg>
```
</details>

Actual | Expected
--- | ---
![actual](https://github.com/weisJ/jsvg/assets/1247730/b4137629-0d40-4d63-a75c-a4044f679c53) | ![expected](https://github.com/weisJ/jsvg/assets/1247730/85f97894-5d5d-4139-98e5-ebacbcd49ae0)

I speculate it is a kind of optimization (related to #65 ?), as artificially extending the object bounds, including a bigger fully transparent geometry, works around the problem:

```xml
  <use href="#graphic2" filter="url(#drop-shadow)" mask="url(#graphic1-mask)" />

  <!--circle id="graphic2" cx="120" cy="110" r="70" fill="orange" /-->
  <g id="graphic2">
    <rect x="50" y="40" width="160" height="160" opacity="0" />
    <circle cx="120" cy="110" r="70" fill="orange" />
  </g>
```

I had initially tried changing the `width` and `height` of filters and masks to much > 100% but observed no effect.

**Comment by weisJ** [comment]
Should be fixed in current snapshot

**Comment by stanio** [comment]
Tested with current 1.5.1-SNAPSHOT – fixed.  Thank you very much!

**Comment by weisJ** [comment]
Now available in 1.6.0
